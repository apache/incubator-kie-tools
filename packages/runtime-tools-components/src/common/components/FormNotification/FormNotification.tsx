/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import React, { useState } from "react";
import { Alert, AlertActionCloseButton, AlertActionLink } from "@patternfly/react-core/dist/js/components/Alert";
import { componentOuiaProps, OUIAProps } from "../../ouiaTools";

export interface Notification {
  type: "success" | "error";
  message: string;
  details?: string;
  customActions?: Action[];
  close: () => void;
}

export interface Action {
  label: string;
  onClick: () => void;
}

interface IOwnProps {
  notification: Notification;
}

export const FormNotification: React.FC<IOwnProps & OUIAProps> = ({ notification, ouiaId, ouiaSafe }) => {
  const variant = notification.type === "error" ? "danger" : "success";

  const [showDetails, setShowDetails] = useState<boolean>(false);
  return (
    <Alert
      isInline
      title={notification.message}
      variant={variant}
      actionLinks={
        <React.Fragment>
          {notification.details && (
            <AlertActionLink onClick={() => setShowDetails(!showDetails)}>View details</AlertActionLink>
          )}
          {notification.customActions &&
            notification.customActions.length > 0 &&
            notification.customActions.map((customAction: Action, index: number) => {
              return (
                <AlertActionLink onClick={customAction.onClick} key={index}>
                  {customAction.label}
                </AlertActionLink>
              );
            })}
        </React.Fragment>
      }
      actionClose={<AlertActionCloseButton onClose={notification.close} />}
      {...componentOuiaProps(ouiaId, "form-notification-alert", ouiaSafe)}
    >
      {showDetails && notification.details && <p>{notification.details}</p>}
    </Alert>
  );
};
