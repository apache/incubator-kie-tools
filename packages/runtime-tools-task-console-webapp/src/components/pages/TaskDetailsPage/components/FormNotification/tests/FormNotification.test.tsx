/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
import React from "react";
import FormNotification, { Notification } from "../FormNotification";
import { AlertActionCloseButton, AlertActionLink, Alert } from "@patternfly/react-core";
import { mount } from "enzyme";

describe("FormNotification test", () => {
  it("Simple notification", () => {
    const notificationProps: Notification = {
      type: "success",
      message: "The form has been submitted",
      close: jest.fn(),
    };

    const wrapper = mount(<FormNotification notification={notificationProps} />).find("FormNotification");

    expect(wrapper).toMatchSnapshot();

    const alert = wrapper.find(Alert);

    expect(alert.exists()).toBeTruthy();
    expect(alert.props().variant).toBe("success");

    expect(wrapper.html()).toContain(notificationProps.message);

    const button = wrapper.find(AlertActionCloseButton).find("button");

    button.simulate("click");

    expect(notificationProps.close).toBeCalled();
  });

  it("Notification with details", async () => {
    const notificationProps: Notification = {
      type: "error",
      message: "The form has been submitted",
      close: jest.fn(),
      details: "The details here!",
    };

    let wrapper = mount(<FormNotification notification={notificationProps} />).find("FormNotification");
    expect(wrapper).toMatchSnapshot();

    expect(wrapper.html()).toContain(notificationProps.message);

    const alert = wrapper.find(Alert);

    expect(alert.exists()).toBeTruthy();
    expect(alert.props().variant).toBe("danger");

    const button = wrapper.find(AlertActionLink).find("button");

    expect(button.exists()).toBeTruthy();
    expect(button.getDOMNode().innerHTML).toBe("View details");

    button.simulate("click");

    expect(wrapper).toMatchSnapshot();

    wrapper = wrapper.update().find("FormNotification");

    expect(wrapper.html()).toContain(notificationProps.details);
  });

  it("Notification with custom action", async () => {
    const notificationProps: Notification = {
      type: "success",
      message: "The form has been submitted",
      close: jest.fn(),
      customAction: {
        label: "Custom action",
        onClick: jest.fn(),
      },
    };

    const wrapper = mount(<FormNotification notification={notificationProps} />).find("FormNotification");
    expect(wrapper).toMatchSnapshot();

    expect(wrapper.html()).toContain(notificationProps.message);

    const button = wrapper.find(AlertActionLink).find("button");

    expect(button.exists()).toBeTruthy();
    expect(button.getDOMNode().innerHTML).toBe("Custom action");

    button.simulate("click");

    expect(notificationProps.customAction.onClick).toBeCalled();
  });
});
