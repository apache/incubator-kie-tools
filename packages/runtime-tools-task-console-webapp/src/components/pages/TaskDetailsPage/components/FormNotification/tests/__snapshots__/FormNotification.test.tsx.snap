// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormNotification test Notification with custom action 1`] = `
<FormNotification
  notification={
    Object {
      "close": [MockFunction],
      "customAction": Object {
        "label": "Custom action",
        "onClick": [MockFunction],
      },
      "message": "The form has been submitted",
      "type": "success",
    }
  }
>
  <Alert
    actionClose={
      <AlertActionCloseButton
        onClose={[MockFunction]}
      />
    }
    actionLinks={
      <React.Fragment>
        <AlertActionLink
          onClick={[MockFunction]}
        >
          Custom action
        </AlertActionLink>
      </React.Fragment>
    }
    data-ouia-component-type="form-notification-alert"
    data-ouia-safe={true}
    isInline={true}
    title="The form has been submitted"
    variant="success"
  >
    <div
      aria-label="Success Alert"
      className="pf-c-alert pf-m-inline pf-m-success"
      data-ouia-component-id="OUIA-Generated-Alert-success-2"
      data-ouia-component-type="form-notification-alert"
      data-ouia-safe={true}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <AlertIcon
        variant="success"
      >
        <div
          className="pf-c-alert__icon"
        >
          <CheckCircleIcon
            color="currentColor"
            noVerticalAlign={false}
            size="sm"
          >
            <svg
              aria-hidden={true}
              aria-labelledby={null}
              fill="currentColor"
              height="1em"
              role="img"
              style={
                Object {
                  "verticalAlign": "-0.125em",
                }
              }
              viewBox="0 0 512 512"
              width="1em"
            >
              <path
                d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
              />
            </svg>
          </CheckCircleIcon>
        </div>
      </AlertIcon>
      <h4
        className="pf-c-alert__title"
      >
        <span
          className="pf-u-screen-reader"
        >
          Success alert:
        </span>
        The form has been submitted
      </h4>
      <div
        className="pf-c-alert__action"
      >
        <AlertActionCloseButton
          onClose={[MockFunction]}
        >
          <Button
            aria-label="Close Success alert: alert: The form has been submitted"
            onClick={[MockFunction]}
            variant="plain"
          >
            <ButtonBase
              aria-label="Close Success alert: alert: The form has been submitted"
              innerRef={null}
              onClick={[MockFunction]}
              variant="plain"
            >
              <button
                aria-disabled={false}
                aria-label="Close Success alert: alert: The form has been submitted"
                className="pf-c-button pf-m-plain"
                data-ouia-component-id="OUIA-Generated-Button-plain-3"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe={true}
                disabled={false}
                onClick={[MockFunction]}
                role={null}
                type="button"
              >
                <TimesIcon
                  color="currentColor"
                  noVerticalAlign={false}
                  size="sm"
                >
                  <svg
                    aria-hidden={true}
                    aria-labelledby={null}
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style={
                      Object {
                        "verticalAlign": "-0.125em",
                      }
                    }
                    viewBox="0 0 352 512"
                    width="1em"
                  >
                    <path
                      d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                    />
                  </svg>
                </TimesIcon>
              </button>
            </ButtonBase>
          </Button>
        </AlertActionCloseButton>
      </div>
      <div
        className="pf-c-alert__action-group"
      >
        <AlertActionLink
          onClick={[MockFunction]}
        >
          <Button
            className=""
            isInline={true}
            onClick={[MockFunction]}
            variant="link"
          >
            <ButtonBase
              className=""
              innerRef={null}
              isInline={true}
              onClick={[MockFunction]}
              variant="link"
            >
              <button
                aria-disabled={false}
                aria-label={null}
                className="pf-c-button pf-m-link pf-m-inline"
                data-ouia-component-id="OUIA-Generated-Button-link-2"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe={true}
                disabled={false}
                onClick={[MockFunction]}
                role={null}
                type="button"
              >
                Custom action
              </button>
            </ButtonBase>
          </Button>
        </AlertActionLink>
      </div>
    </div>
  </Alert>
</FormNotification>
`;

exports[`FormNotification test Notification with details 1`] = `
<FormNotification
  notification={
    Object {
      "close": [MockFunction],
      "details": "The details here!",
      "message": "The form has been submitted",
      "type": "error",
    }
  }
>
  <Alert
    actionClose={
      <AlertActionCloseButton
        onClose={[MockFunction]}
      />
    }
    actionLinks={
      <React.Fragment>
        <AlertActionLink
          onClick={[Function]}
        >
          View details
        </AlertActionLink>
      </React.Fragment>
    }
    data-ouia-component-type="form-notification-alert"
    data-ouia-safe={true}
    isInline={true}
    title="The form has been submitted"
    variant="danger"
  >
    <div
      aria-label="Danger Alert"
      className="pf-c-alert pf-m-inline pf-m-danger"
      data-ouia-component-id="OUIA-Generated-Alert-danger-1"
      data-ouia-component-type="form-notification-alert"
      data-ouia-safe={true}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <AlertIcon
        variant="danger"
      >
        <div
          className="pf-c-alert__icon"
        >
          <ExclamationCircleIcon
            color="currentColor"
            noVerticalAlign={false}
            size="sm"
          >
            <svg
              aria-hidden={true}
              aria-labelledby={null}
              fill="currentColor"
              height="1em"
              role="img"
              style={
                Object {
                  "verticalAlign": "-0.125em",
                }
              }
              viewBox="0 0 512 512"
              width="1em"
            >
              <path
                d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
              />
            </svg>
          </ExclamationCircleIcon>
        </div>
      </AlertIcon>
      <h4
        className="pf-c-alert__title"
      >
        <span
          className="pf-u-screen-reader"
        >
          Danger alert:
        </span>
        The form has been submitted
      </h4>
      <div
        className="pf-c-alert__action"
      >
        <AlertActionCloseButton
          onClose={[MockFunction]}
        >
          <Button
            aria-label="Close Danger alert: alert: The form has been submitted"
            onClick={[MockFunction]}
            variant="plain"
          >
            <ButtonBase
              aria-label="Close Danger alert: alert: The form has been submitted"
              innerRef={null}
              onClick={[MockFunction]}
              variant="plain"
            >
              <button
                aria-disabled={false}
                aria-label="Close Danger alert: alert: The form has been submitted"
                className="pf-c-button pf-m-plain"
                data-ouia-component-id="OUIA-Generated-Button-plain-2"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe={true}
                disabled={false}
                onClick={[MockFunction]}
                role={null}
                type="button"
              >
                <TimesIcon
                  color="currentColor"
                  noVerticalAlign={false}
                  size="sm"
                >
                  <svg
                    aria-hidden={true}
                    aria-labelledby={null}
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style={
                      Object {
                        "verticalAlign": "-0.125em",
                      }
                    }
                    viewBox="0 0 352 512"
                    width="1em"
                  >
                    <path
                      d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                    />
                  </svg>
                </TimesIcon>
              </button>
            </ButtonBase>
          </Button>
        </AlertActionCloseButton>
      </div>
      <div
        className="pf-c-alert__action-group"
      >
        <AlertActionLink
          onClick={[Function]}
        >
          <Button
            className=""
            isInline={true}
            onClick={[Function]}
            variant="link"
          >
            <ButtonBase
              className=""
              innerRef={null}
              isInline={true}
              onClick={[Function]}
              variant="link"
            >
              <button
                aria-disabled={false}
                aria-label={null}
                className="pf-c-button pf-m-link pf-m-inline"
                data-ouia-component-id="OUIA-Generated-Button-link-1"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe={true}
                disabled={false}
                onClick={[Function]}
                role={null}
                type="button"
              >
                View details
              </button>
            </ButtonBase>
          </Button>
        </AlertActionLink>
      </div>
    </div>
  </Alert>
</FormNotification>
`;

exports[`FormNotification test Notification with details 2`] = `
<FormNotification
  notification={
    Object {
      "close": [MockFunction],
      "details": "The details here!",
      "message": "The form has been submitted",
      "type": "error",
    }
  }
>
  <Alert
    actionClose={
      <AlertActionCloseButton
        onClose={[MockFunction]}
      />
    }
    actionLinks={
      <React.Fragment>
        <AlertActionLink
          onClick={[Function]}
        >
          View details
        </AlertActionLink>
      </React.Fragment>
    }
    data-ouia-component-type="form-notification-alert"
    data-ouia-safe={true}
    isInline={true}
    title="The form has been submitted"
    variant="danger"
  >
    <div
      aria-label="Danger Alert"
      className="pf-c-alert pf-m-inline pf-m-danger"
      data-ouia-component-id="OUIA-Generated-Alert-danger-1"
      data-ouia-component-type="form-notification-alert"
      data-ouia-safe={true}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <AlertIcon
        variant="danger"
      >
        <div
          className="pf-c-alert__icon"
        >
          <ExclamationCircleIcon
            color="currentColor"
            noVerticalAlign={false}
            size="sm"
          >
            <svg
              aria-hidden={true}
              aria-labelledby={null}
              fill="currentColor"
              height="1em"
              role="img"
              style={
                Object {
                  "verticalAlign": "-0.125em",
                }
              }
              viewBox="0 0 512 512"
              width="1em"
            >
              <path
                d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
              />
            </svg>
          </ExclamationCircleIcon>
        </div>
      </AlertIcon>
      <h4
        className="pf-c-alert__title"
      >
        <span
          className="pf-u-screen-reader"
        >
          Danger alert:
        </span>
        The form has been submitted
      </h4>
      <div
        className="pf-c-alert__action"
      >
        <AlertActionCloseButton
          onClose={[MockFunction]}
        >
          <Button
            aria-label="Close Danger alert: alert: The form has been submitted"
            onClick={[MockFunction]}
            variant="plain"
          >
            <ButtonBase
              aria-label="Close Danger alert: alert: The form has been submitted"
              innerRef={null}
              onClick={[MockFunction]}
              variant="plain"
            >
              <button
                aria-disabled={false}
                aria-label="Close Danger alert: alert: The form has been submitted"
                className="pf-c-button pf-m-plain"
                data-ouia-component-id="OUIA-Generated-Button-plain-2"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe={true}
                disabled={false}
                onClick={[MockFunction]}
                role={null}
                type="button"
              >
                <TimesIcon
                  color="currentColor"
                  noVerticalAlign={false}
                  size="sm"
                >
                  <svg
                    aria-hidden={true}
                    aria-labelledby={null}
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style={
                      Object {
                        "verticalAlign": "-0.125em",
                      }
                    }
                    viewBox="0 0 352 512"
                    width="1em"
                  >
                    <path
                      d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                    />
                  </svg>
                </TimesIcon>
              </button>
            </ButtonBase>
          </Button>
        </AlertActionCloseButton>
      </div>
      <div
        className="pf-c-alert__action-group"
      >
        <AlertActionLink
          onClick={[Function]}
        >
          <Button
            className=""
            isInline={true}
            onClick={[Function]}
            variant="link"
          >
            <ButtonBase
              className=""
              innerRef={null}
              isInline={true}
              onClick={[Function]}
              variant="link"
            >
              <button
                aria-disabled={false}
                aria-label={null}
                className="pf-c-button pf-m-link pf-m-inline"
                data-ouia-component-id="OUIA-Generated-Button-link-1"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe={true}
                disabled={false}
                onClick={[Function]}
                role={null}
                type="button"
              >
                View details
              </button>
            </ButtonBase>
          </Button>
        </AlertActionLink>
      </div>
    </div>
  </Alert>
</FormNotification>
`;

exports[`FormNotification test Simple notification 1`] = `
<FormNotification
  notification={
    Object {
      "close": [MockFunction],
      "message": "The form has been submitted",
      "type": "success",
    }
  }
>
  <Alert
    actionClose={
      <AlertActionCloseButton
        onClose={[MockFunction]}
      />
    }
    actionLinks={<React.Fragment />}
    data-ouia-component-type="form-notification-alert"
    data-ouia-safe={true}
    isInline={true}
    title="The form has been submitted"
    variant="success"
  >
    <div
      aria-label="Success Alert"
      className="pf-c-alert pf-m-inline pf-m-success"
      data-ouia-component-id="OUIA-Generated-Alert-success-1"
      data-ouia-component-type="form-notification-alert"
      data-ouia-safe={true}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <AlertIcon
        variant="success"
      >
        <div
          className="pf-c-alert__icon"
        >
          <CheckCircleIcon
            color="currentColor"
            noVerticalAlign={false}
            size="sm"
          >
            <svg
              aria-hidden={true}
              aria-labelledby={null}
              fill="currentColor"
              height="1em"
              role="img"
              style={
                Object {
                  "verticalAlign": "-0.125em",
                }
              }
              viewBox="0 0 512 512"
              width="1em"
            >
              <path
                d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
              />
            </svg>
          </CheckCircleIcon>
        </div>
      </AlertIcon>
      <h4
        className="pf-c-alert__title"
      >
        <span
          className="pf-u-screen-reader"
        >
          Success alert:
        </span>
        The form has been submitted
      </h4>
      <div
        className="pf-c-alert__action"
      >
        <AlertActionCloseButton
          onClose={[MockFunction]}
        >
          <Button
            aria-label="Close Success alert: alert: The form has been submitted"
            onClick={[MockFunction]}
            variant="plain"
          >
            <ButtonBase
              aria-label="Close Success alert: alert: The form has been submitted"
              innerRef={null}
              onClick={[MockFunction]}
              variant="plain"
            >
              <button
                aria-disabled={false}
                aria-label="Close Success alert: alert: The form has been submitted"
                className="pf-c-button pf-m-plain"
                data-ouia-component-id="OUIA-Generated-Button-plain-1"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe={true}
                disabled={false}
                onClick={[MockFunction]}
                role={null}
                type="button"
              >
                <TimesIcon
                  color="currentColor"
                  noVerticalAlign={false}
                  size="sm"
                >
                  <svg
                    aria-hidden={true}
                    aria-labelledby={null}
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style={
                      Object {
                        "verticalAlign": "-0.125em",
                      }
                    }
                    viewBox="0 0 352 512"
                    width="1em"
                  >
                    <path
                      d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                    />
                  </svg>
                </TimesIcon>
              </button>
            </ButtonBase>
          </Button>
        </AlertActionCloseButton>
      </div>
      <div
        className="pf-c-alert__action-group"
      />
    </div>
  </Alert>
</FormNotification>
`;
